lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1823,10 +1823,18 @@ sub git_log_diff ($self, $dir, $refspec_range, $limit = undef) {
     return $res->{stdout} . $res->{stderr};
 }
 
-sub git_diff ($self, $dir, $refspec_range) {
+sub git_diff ($self, $dir, $refspec_range, $limit = undef) {
     return "Invalid range $refspec_range" if $refspec_range =~ m/UNKNOWN|unreadable git hash/;
     my $timeout = OpenQA::App->singleton->config->{global}->{job_investigate_git_timeout} // 20;
-    my $res = run_cmd_with_log_return_error(['timeout', $timeout, 'git', '-C', $dir, 'diff', '--stat', $refspec_range],
+    my $res = run_cmd_with_log_return_error(['git', '-C', $dir, 'rev-list', '--count', $refspec_range]);
+    if ($res->{return_code}) {
+        return $res->{stdout} . $res->{stderr};
+    }
+    chomp(my $count = $res->{stdout});
+    if ($count > $limit) {
```

Comment:
Maybe it would still make sense to check whether we actually got a number here. After all, this is input from an *external* process and thus deserves basic validation.

Reply:
Considering that we would have gotten a warning in the log (and an email in our inbox) if `git rev-list --count` printed something non-numeric while returning `0`, I didn't consider this really important.
It would mean that git is broken, or our subroutine wrapping the git call is broken, and an entry in our log would be the right thing.

Now I thought, ok, if I shall check if `$count` is numeric, then I should actually do proper error handling for everything.
I'm now checking everything I can in `git_diff`, and issue a warning with the git output if something is unexpected, while the user gets an unspecific error message that doesn't reveal possible internal details.